oxidation/libparsec/crates/platform_async/src/native/join_set.rs

```diff
@@ -42,6 +42,9 @@ pub struct JoinSet<T> {
 }
 
 impl<T> JoinSet<T> {
+    pub fn new() -> Self {
+        Self { tasks: vec![] }
+    }
```

Comment:
see https://github.com/Scille/parsec-cloud/pull/4258#discussion_r1143061038

Reply:
https://github.com/Scille/parsec-cloud/pull/4258#discussion_r1143085075 :p

Reply:
Impl `Default` instead ?

Reply:
Nop, impl default on T means the issue (https://github.com/Scille/parsec-cloud/pull/4258#discussion_r1143126758) will occurs every time we use JoinSet (because we won't implement Default on all data) :(
    
`JoinSet::new()` is just fine